### PR TITLE
Move to async with tokio 0.2

### PR DESCRIPTION
**Pull request by clux**
This is to move to the async ecosystem to get:

- reqwest 0.10
- kube (like the last 8 versions)

and to be able to rely on async stream mechanics for parallelism rather than forcing rayon to do it badly.

This is actually performing really well so far for `shipcat cluster diff` and `shipcat cluster crd reconcile`, so might just have this as a start. The rest can be asyncified at will.

### CODE REVIEW

**Comment by clux** [comment]
replaces #384 because async_std was more awkward to deal with.

fixes #395 and #394

**Comment by clux** [comment]
Going to leave this as is without trying to unify the reqwest versions (which will reduce the dependency tree quite a bit), because the PR is getting quite big.

File: shipcat_cli/src/top.rs [hunk 1]
```diff
@@ -38,51 +38,66 @@ impl FromStr for ResourceOrder {
...
-            }
-            if let Some(mf) = first_mf {
-                Ok(Some((mf, res)))
-            } else {
-                Ok(None)
+async fn calculate_manifest_requests(conf: &Config, reg: &Region) -> Result<Vec<(Manifest, ResourceTotals)>> {
+    let available = shipcat_filebacked::available(conf, &reg).await?;
+    let mut buffered = stream::iter(available)
+        .map(move |mf| load_mf_req(mf.base.name, conf, reg))
+        .buffer_unordered(100);
```
Comment (bh-anton-k) [inline shipcat_cli/src/top.rs]: 100? 🤔I though it should be bound by core count, practical value related to concurrent TCP connections/in-flight requests we expect to be keeping to kube backend or something, I dunno, looks like a magick number to me

Comment (clux) [reply shipcat_cli/src/top.rs]: Yeah, I am fixing this in the next pr

**Pull request merged**